File: DependencyInjection/DependencyInjectionAfter/_1_ControlFreakRefactoredToDependencyInjection/Core/TeleComSystem.cs
using DependencyInjectionAfter._1_ControlFreakRefactoredToDependencyInjection.Inbound;
using DependencyInjectionAfter._1_ControlFreakRefactoredToDependencyInjection.Outbound;

namespace DependencyInjectionAfter._1_ControlFreakRefactoredToDependencyInjection.Core
{
  class TeleComSystem
  {
    private readonly IAcmeProcessingWorkflow _processingWorkflow;
    private readonly IInbound _inbound;
    private readonly IOutbound _outbound;

    public TeleComSystem(
      IInbound binaryUdpInbound, 
      IOutbound xmlTcpOutbound, 
      IAcmeProcessingWorkflow acmeProcessingWorkflow)
    {
      _inbound = binaryUdpInbound;
      _outbound = xmlTcpOutbound;
      _processingWorkflow = acmeProcessingWorkflow;
    }

    public void Start()
    {
      _inbound.SetDomainLogic(_processingWorkflow);
      _processingWorkflow.SetOutbound(_outbound);
      _inbound.StartListening();
    }
  }
}

File: Java/dependency-injection/_02_control-freak/src/main/java/com/github/grzesiek_galezowski/DependencyInjection/_1_ControlFreak/Core/TeleComSystem.java
package com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.Core;

import com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.Inbound.BinaryUdpInbound;
import com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.Outbound.XmlTcpOutbound;

public class TeleComSystem {
    private final AcmeProcessingWorkflow processingWorkflow;
    private final BinaryUdpInbound inbound;
    private final XmlTcpOutbound outbound;

    public TeleComSystem() {
      inbound = new BinaryUdpInbound();
      outbound = new XmlTcpOutbound();
      processingWorkflow = new AcmeProcessingWorkflow();
    }

    public void start() {
      inbound.setDomainLogic(processingWorkflow);
      processingWorkflow.setOutbound(outbound);
      inbound.startListening();
    }

  }